lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -338,13 +346,24 @@ impl<C: CollectionContainer> ConsensusState<C> {
         operation: ConsensusOperations,
         wait_timeout: Option<Duration>,
     ) -> Result<bool, StorageError> {
+        let wait_timeout = wait_timeout.unwrap_or(DEFAULT_META_OP_WAIT);
+
+        if !self
+            .is_leader_established
+            .await_ready_for_timeout(wait_timeout)
```

Comment:
the downside of this is that the thread is blocked. But consensus operations are expected to be rare and they should not affect the performance significantly